src/contexts/I18n/index.js

```diff
@@ -13,35 +14,25 @@ export function useI18n() {
 }
 
 function I18nProvider(props) {
-  const { children, defaultLocale = 'en', locale } = props
+  const { children, defaultLocale = 'en', locale = defaultLocale } = props
 
   let translations
   try {
     // eslint-disable-next-line global-require, import/no-dynamic-require
-    translations = require(`../../../public/locales/${locale}.json`)
+    const { messages } = require(`@lingui/loader!../../../public/locales/${locale}/messages.po`)
+    translations = messages
   } catch (err) {
-    // eslint-disable-next-line global-require, import/no-dynamic-require
-    translations = require(`../../../public/locales/${defaultLocale}.json`)
+    const {
+      messages,
+      // eslint-disable-next-line global-require, import/no-dynamic-require
+    } = require(`@lingui/loader!../../../public/locales/${defaultLocale}/messages.po`)
+    translations = messages
   }
 
-  i18nConfig({
-    translations,
-    // Intl DateTimeFormat options as described here: https://goo.gl/lslekB
-    date: {},
-    // Intl NumberFormat options as described here: https://goo.gl/pDwbG2
-    number: {},
-  })
-
-  const contextValue = React.useMemo(
-    () => ({
-      defaultLocale,
-      locale,
-      t: i18n,
-    }),
-    [defaultLocale, locale],
-  )
-
-  return <I18nContext.Provider value={contextValue}>{children}</I18nContext.Provider>
+  i18n.load(locale, translations)
+  i18n.activate(locale)
```

Comment:
I'd say that this solution *works*. Maybe we can wrap it in a `useEffect`? 

My biggest question is regarding what impact it would have by wrapping it in a `useEffect` [as in `next.js/examples/with-lingui`](https://github.com/vercel/next.js/blob/2abc824869e4927b820767c1c416a381c8ae528d/examples/with-lingui/pages/_app.js#L15-L24)

Reply:
Let's try it out when we have the chance :)

Reply:
Example on nextjs side is broken. Please use examples from lingui repo. With example from nextjs repo you will never get SSR working.